source/reference/compatibility.txt

```diff
@@ -202,6 +202,30 @@ in ``mongosh`` better align with the types used by the MongoDB Drivers.
    For more information on managing types, refer to the
    :manual:`schema validation overview </core/schema-validation>`.
 
+Undefined Values
+----------------
+
+The undefined BSON type is `deprecated <https://bsonspec.org/spec.html>`__ in
+:binary:`bin.mongosh`. If you insert a document with the undefined JS type in
+:binary:`bin.mongosh`, your deployment replaces the undefined value in your 
+document with the BSON null value. There is no supported way of inserting undefined
+values into your database using :binary:`bin.mongosh`.
```

Comment:
It‘s also relevant that mongosh will represent any stored BSON undefined values as null. So if you’ve stored a BSON undefined value with some other tool—e.g., Go’s driver or the `mongo` shell—then you fetch that document with mongosh, you’ll see null where the actual stored document has undefined.

Should this section mention that as well?